src/rust/src/x509/certificate.rs

```diff
@@ -418,24 +418,38 @@ fn warn_if_negative_serial(py: pyo3::Python<'_>, bytes: &'_ [u8]) -> pyo3::PyRes
 
 fn warn_if_invalid_ecdsa_params(
     py: pyo3::Python<'_>,
-    signature_algorithm: AlgorithmIdentifier<'_>,
+    owned_certificate: &OwnedCertificate,
 ) -> pyo3::PyResult<()> {
-    match signature_algorithm.params {
-        AlgorithmParameters::EcDsaWithSha224(Some(..))
-        | AlgorithmParameters::EcDsaWithSha256(Some(..))
-        | AlgorithmParameters::EcDsaWithSha384(Some(..))
-        | AlgorithmParameters::EcDsaWithSha512(Some(..)) => {
-            let cryptography_warning = py
-                .import(pyo3::intern!(py, "cryptography.utils"))?
-                .getattr(pyo3::intern!(py, "DeprecatedIn41"))?;
-            pyo3::PyErr::warn(
-                py,
-                cryptography_warning,
-                "The parsed certificate contains a NULL parameter value in its signature algorithm parameters. This is invalid and will be rejected in a future version of cryptography. If this certificate was created via Java, please upgrade to JDK16+ or the latest JDK11 once a fix is issued. If this certificate was created in some other fashion please report the issue to the cryptography issue tracker. See https://github.com/pyca/cryptography/issues/8996 for more details.",
-                1,
-            )?;
+    for params in [
+        owned_certificate
+            .borrow_value()
+            .signature_alg
+            .params
+            .clone(),
+        owned_certificate
+            .borrow_value()
+            .tbs_cert
+            .signature_alg
+            .params
+            .clone(),
+    ] {
```

Comment:
This is way too convoluted. Just use the function from the previous commit and call it 2x, no need for this for loop.